trieste/acquisition/rule.py

```diff
@@ -1103,6 +1103,275 @@ def state_func(
+        self._rule = rule
+
+    def __repr__(self) -> str:
+        """"""
+        return f"""{self.__class__.__name__}({self._init_subspaces!r}, {self._rule!r})"""
+
+    def acquire(
+        self,
+        search_space: SearchSpace,
+        models: Mapping[Tag, ProbabilisticModelType],
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> types.State[State | None, TensorType]:
+        def state_func(
+            state: MultiTrustRegion.State | None,
+        ) -> Tuple[MultiTrustRegion.State | None, TensorType]:
+            """If state is None, initialise the subspaces by picking new locations. Otherwise,
+            update the existing subspaces.
+            Initialize the subspaces if necessary, potentially looking at the entire group.
+            Use the rule to acquire points from the acquisition space.
+            """
+            # If state is set, the tags should be the same as the tags of the acquisition space
+            # in the state.
+            if state is not None:
+                assert (
+                    self._tags == state.acquisition_space.subspace_tags
+                ), f"""The tags of the state acquisition space
+                    {state.acquisition_space.subspace_tags} should be the same as the tags of the
+                    MultiTrustRegion acquisition rule {self._tags}"""
+
+            subspaces = []
+            for index, tag in enumerate(self._tags):
+                if state is None:
+                    subspace = self._init_subspaces[index]
+                    subspace.initialize(models, datasets)
+                else:
+                    _subspace = state.acquisition_space.get_subspace(tag)
+                    assert isinstance(_subspace, type(self._init_subspaces[index]))
+                    subspace = _subspace
+                    subspace.update(models, datasets)
+
+                subspaces.append(subspace)
+
+            self.maybe_initialize_subspaces(subspaces, models, datasets)
+
+            if state is None:
+                acquisition_space = TaggedMultiSearchSpace(subspaces, self._tags)
+            else:
+                acquisition_space = state.acquisition_space
+
+            state_ = MultiTrustRegion.State(acquisition_space)
+            points = self._rule.acquire(acquisition_space, models, datasets=datasets)
+
+            return state_, points
+
+        return state_func
+
+    def maybe_initialize_subspaces(
+        self,
+        subspaces: Sequence[UpdateableSearchSpaceType],
+        models: Mapping[Tag, ProbabilisticModelType],
+        datasets: Optional[Mapping[Tag, Dataset]] = None,
+    ) -> None:
+        """Initialize subspaces if necessary.
+        Get a mask of subspaces that need to be initialized using an abstract method.
+        Initialize individual subpaces by calling the method of the UpdateableSearchSpaceType class.
+        """
+        mask = self.get_initialize_subspaces_mask(subspaces, models, datasets)
```

Comment:
(maybe add some checking on the mask shape? alternatively you may be able to add shape checking to get_initialize_subspaces_mask)